jetty-core/jetty-util/src/test/java/org/eclipse/jetty/util/resource/MountedPathResourceTest.java

```diff
@@ -362,4 +362,75 @@ public void testJarFileResourceListDirWithSpace() throws Exception
             assertThat("Dir contents", actual, containsInAnyOrder(expected));
         }
     }
+
+    /**
+     * When mounting multiple points within the same JAR, only
+     * 1 mount should be created, but have reference counts
+     * tracked separately.  Done via {@link ResourceFactory.Closeable}
+     * essentially a duplicate of {@link #testMountByJarNameLifeCycle()}
+     * to ensure parity in the two implementations.
+     */
+    @Test
+    public void testMountByJarNameClosable()
+    {
+        Path jarPath = MavenPaths.findTestResourceFile("jar-file-resource.jar");
+        URI uriRoot = URI.create("jar:" + jarPath.toUri().toASCIIString() + "!/"); // root
+        URI uriRez = URI.create("jar:" + jarPath.toUri().toASCIIString() + "!/rez/"); // dir
+        URI uriDeep = URI.create("jar:" + jarPath.toUri().toASCIIString() + "!/rez/deep/"); // dir
+        URI uriZzz = URI.create("jar:" + jarPath.toUri().toASCIIString() + "!/rez/deep/zzz"); // file
+
+        try (ResourceFactory.Closeable resourceFactory = ResourceFactory.closeable())
+        {
+            Resource resRoot = resourceFactory.newResource(uriRoot);
+            Resource resRez = resourceFactory.newResource(uriRez);
+            Resource resDeep = resourceFactory.newResource(uriDeep);
+            Resource resZzz = resourceFactory.newResource(uriZzz);
+
+            assertThat(FileSystemPool.INSTANCE.mounts().size(), is(1));
+            int mountCount = FileSystemPool.INSTANCE.getReferenceCount(uriRoot);
+            assertThat(mountCount, is(4));
+        }
```

Comment:
You should also assert mount count == 0 and getReferenceCount(uriRoot) == 0  after the `resourceFactory` is closed.

Reply:
Done